server/src/main/java/com/flipkart/varadhi/components/Component.java

```diff
@@ -0,0 +1,12 @@
+package com.flipkart.varadhi.components;
+
+import com.flipkart.varadhi.CoreServices;
+import com.flipkart.varadhi.cluster.ClusterManager;
+import com.flipkart.varadhi.config.AppConfiguration;
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
+
+public interface Component {
```

Comment:
it feels redundant to define this base component interface. 

Can't  we make the server/controller/consumer "verticles"? The verticles have lifecycle management and are connected among themselves via event bus.